packages/guides/src/Compiler/NodeTransformers/MoveAnchorTransformer.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\NodeTransformers;
+
+use ArrayIterator;
+use LogicException;
+use phpDocumentor\Guides\Compiler\CompilerContext;
+use phpDocumentor\Guides\Compiler\NodeTransformer;
+use phpDocumentor\Guides\Compiler\ShadowTree\TreeNode;
+use phpDocumentor\Guides\Nodes\AnchorNode;
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\Nodes\SectionNode;
+
+use function spl_object_hash;
+
+/** @implements NodeTransformer<AnchorNode> */
+final class MoveAnchorTransformer implements NodeTransformer
+{
+    /** @var array<string, string> */
+    private array $seen = [];
+
+    public function enterNode(Node $node, CompilerContext $compilerContext): Node
+    {
+        return $node;
+    }
+
+    public function leaveNode(Node $node, CompilerContext $compilerContext): Node|null
+    {
+        //When exists in seen, it means that the node has already been processed. Ignore it.
+        if (isset($this->seen[spl_object_hash($node)])) {
+            return $node;
+        }
+
+        $this->seen[spl_object_hash($node)] = spl_object_hash($node);
+        $parent = $compilerContext->getShadowTree()->getParent();
+        if ($parent === null) {
+            throw new LogicException('Node not found in shadow tree');
+        }
+
+        $position = $parent->findPosition($node);
+        if ($position === null) {
+            throw new LogicException('Node not found in shadow tree');
+        }
+
+        return $this->attemptMoveToNeighbour($parent, $position, $node);
+    }
+
+    public function supports(Node $node): bool
+    {
+        return $node instanceof AnchorNode;
+    }
+
+    public function getPriority(): int
+    {
+        return 30000;
```

Comment:
```suggestion
        return 30_000;
```

Reply:
Thanks, I added this as a style rule to php-code-sniffer